Docker container build fails when building Python 3.7

Build of the Celery docker container fails when running `docker-compose build celery` as described in the [Developing and Testing with Docker](http://docs.celeryproject.org/en/master/contributing.html#developing-and-testing-with-docker) section of the _Contributing_ documentation.

Build environment:
Host OS: Ubuntu 16.04
Docker version: 18.09.0
Docker-compose version: 1.23.2 (Python 3.6.7)

During build of the Celery container, the build fails with:

```
...
Downloading Python-3.7.1.tar.xz...
-> https://www.python.org/ftp/python/3.7.1/Python-3.7.1.tar.xz
Installing Python-3.7.1...
ERROR: The Python ssl extension was not compiled. Missing the OpenSSL lib?

Please consult to the Wiki page to fix the problem.
https://github.com/pyenv/pyenv/wiki/Common-build-problems


BUILD FAILED (Debian 8.11 using python-build 20180424)
...
```

It looks like this is a result of the default packaged version of `openssl`/`libssl` in Debian "jessie" being 1.0.1. Python 3.7 seems to require OpenSSL >= 1.0.2.

I resolved this by updating the `Dockerfile` to install the OpenSSL 1.0.2 packages in the [jessie-backports](https://packages.debian.org/jessie-backports/openssl) repo. PR for this update to follow shortly, however, maybe a preferable approach is to update the base container to [Debian Stretch](https://www.debian.org/releases/stable/)?

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -27,6 +27,10 @@ RUN apt-get update && apt-get install -y build-essential \
     zlib1g-dev \
     pkg-config
 
+# Update libssl to 1.0.2 from backports to support Python 3.7
+RUN echo "deb http://deb.debian.org/debian jessie-backports main" | tee /etc/apt/sources.list.d/jessie-backports.list
+RUN apt-get update && apt-get install -y -t jessie-backports libssl-dev
+
 # Setup variables. Even though changing these may cause unnecessary invalidation of
 # unrelated elements, grouping them together makes the Dockerfile read better.
 ENV PROVISIONING /provisioning

